### PR TITLE
[Merged by Bors] - feat: notation and delaborator for inner product

### PR DESCRIPTION
**Pull request by astrainfinita**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/InnerProductSpace/Basic.lean [hunk 1]
```diff
@@ -81,11 +81,18 @@ class Inner (𝕜 E : Type*) where
...
+
+open Lean PrettyPrinter.Delaborator SubExpr in
+/-- Delaborator for `Inner.inner` -/
+@[delab app.Inner.inner]
+def delabInner : Delab := do
+  guard <| (← getExpr).isAppOfArity' ``inner 5
+  let 𝕜 ← withNaryArg 0 delab
+  let x ← withNaryArg 3 delab
+  let y ← withNaryArg 4 delab
+  `(⟪$x, $y⟫_$𝕜)
```
Comment (eric-wieser) [inline Mathlib/Analysis/InnerProductSpace/Basic.lean:95]: Untested:
```suggestion
/-- The inner product with values in `𝕜`. -/
notation3:max "⟪" x ", " y "⟫_" 𝕜:max => @inner 𝕜 _ _ x y
```

Comment (negiizhao) [reply Mathlib/Analysis/InnerProductSpace/Basic.lean:95]: Should we still use `notation3` now?

Comment (dupuisf) [reply Mathlib/Analysis/InnerProductSpace/Basic.lean:95]: Yes. I think the current plan is to rename `notation3` to something that doesn't look deprecated. In any case, I just tested the `notation3` version and it works fine, I think we should go with that.

**Comment by j-loreaux** [comment]
bors merge

**Pull request closed without merging**